vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java

```diff
@@ -130,6 +135,74 @@ public void performLighthouseScan(ScanType scanType, String webPageUrl,
         }
     }
 
+    /**
+     * Performs a Lighthouse scan on both the baseline and checkpoint pages validating that the audit scores of the
+     * checkpoint page have not worsened compared to the audit scores of the baseline page.
+     *
+     * @param scanType       The scan type to use, either <b>full</b> or <b>desktop</b> or <b>mobile</b>.
+     * @param checkpointPage The checkpoint page.
+     * @param baselinePage   The baseline page.
+     * @throws IOException if an I/O exception of some sort has occurred
+     */
+    @SuppressWarnings("MagicNumber")
+    @Then("Lighthouse $scanType audit scores for `$checkpointPage` page are not less than for `$baselinePage` page")
+    public void performLighthouseScanWithComparison(ScanType scanType, String checkpointPage, String baselinePage)
+            throws IOException
+    {
+        for (String strategy : scanType.getStrategies())
+        {
+            LighthouseResultV5 baseline = executePagespeed(baselinePage, strategy, true);
+            LighthouseResultV5 checkpoint = executePagespeed(checkpointPage, strategy, true);
+
+            Map<String, Integer> checkpointScores = getCategoryScores(checkpoint);
+
+            softAssert.runIgnoringTestFailFast(() -> getCategoryScores(baseline).forEach((categoryKey, baselineScore) ->
+            {
+                Integer checkpointScore = checkpointScores.get(categoryKey);
+
+                if (checkpointScore >= baselineScore)
+                {
+                    softAssert.recordPassedAssertion(String.format(
+                            "[%s] The %s audit is passed as checkpoint score (%s) is not less than baseline score (%s)",
+                            strategy, categoryKey, checkpointScore, baselineScore));
+                    return;
+                }
+
+                int scoreDecrease = (int) Math
+                        .abs(((checkpointScore.doubleValue() - baselineScore) / baselineScore) * 100);
+
+                if (scoreDecrease <= acceptableScorePercentageDelta)
+                {
+                    softAssert.recordPassedAssertion(String.format(
+                            "[%s] The %s audit is passed as checkpoint score (%s) fits acceptable delta in %s percents"
+                            + " from baseline score (%s)",
+                            strategy, categoryKey, checkpointScore, acceptableScorePercentageDelta, baselineScore));
+                    return;
+                }
+
+                softAssert.recordFailedAssertion(String.format("[%s] The %s audit score is decreased on %d percents",
+                        strategy, categoryKey, scoreDecrease));
+            }));
+
+            attachmentPublisher.publishAttachment(
+                    "/allure-customization/webjars/vividus-lighthouse-viewer-adaptation/index.html",
```

Comment:
shouldn't it be FTL?

Reply:
not required

Reply:
just didn't want to add more actions to the build guide, only essential ones